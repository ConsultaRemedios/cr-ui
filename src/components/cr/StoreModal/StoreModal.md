StoreModal is a component the displays the store information in a modal.

```js
new Vue({
  data() {
    return {
      store: {
        name: 'Store example',
        logoPath: 'https://via.placeholder.com/130x74',
        informations: [{
          title: 'Panvel Farmácias',
          content: 'Dimed S/A Distribuidora de Medicamentos<br />CNPJ 92.665.611/0001-77<br />AFE: 7.17094.5 <br />CMVS - 355030801-477-002443-1-7'
        }, {
          title: 'Farmacêutico responsável',
          content: 'Ely Fernandes Lima<br />CRF/PR: 15448'
        }, {
          title: 'Horário de funcionamento',
          content: 'Segunda a Sexta <br />8h00 às 20h30<br />Av. Industrial Belgraf, 865 Eldorado do Sul, Rio Grande do Sul'
        }, {
          title: 'Atendimento',
          content: '(51) 3481-9500<br />sac@grupodimed.com.br'
        }],
      },
      ui: {
        isOpen: false,
      }
    };
  },
  methods: {
    openModal() {
      this.ui.isOpen = true;
    },

    closeModal() {
      this.ui.isOpen = false;
    },
  },
  template: `
    <div>
      <BaseButton @click="openModal">Toggle StoreModal</BaseButton>
      <StoreModal
        v-if="ui.isOpen"
        :name="store.name"
        :logoPath="store.logoPath"
        :informations="store.informations"
        @close="closeModal"
      />
    </div>
  `
})
```

In case you don't have the information about the store yet you can set the loading status and fetch the information.

```js
const storeInfo = {
  name: 'Store example',
  logoPath: 'https://via.placeholder.com/130x74',
  informations: [{
    title: 'Panvel Farmácias',
    content: 'Dimed S/A Distribuidora de Medicamentos<br />CNPJ 92.665.611/0001-77<br />AFE: 7.17094.5 <br />CMVS - 355030801-477-002443-1-7'
  }, {
    title: 'Farmacêutico responsável',
    content: 'Ely Fernandes Lima<br />CRF/PR: 15448'
  }, {
    title: 'Horário de funcionamento',
    content: 'Segunda a Sexta <br />8h00 às 20h30<br />Av. Industrial Belgraf, 865 Eldorado do Sul, Rio Grande do Sul'
  }, {
    title: 'Atendimento',
    content: '(51) 3481-9500<br />sac@grupodimed.com.br'
  }],
};

const makeFetch = () => new Promise(resolve => {
  setTimeout(() => {
    resolve(storeInfo);
  }, 1000);
});

new Vue({
  data() {
    return {
      store: {
        name: '',
        logoPath: '',
        informations: [],
      },
      ui: {
        isLoading: false,
        isOpen: false
      },
    };
  },
  methods: {
    openModal() {
      this.ui.isOpen = true;
      this.ui.isLoading = true;

      makeFetch()
        .then(({ name, logoPath, informations }) => {
          this.store.name = name;
          this.store.logoPath = logoPath;
          this.store.informations = informations;

          this.ui.isLoading = false;
        });
    },

    closeModal() {
      this.ui.isOpen = false;
      this.ui.isLoading = false;
      this.store = {
        name: '',
        logoPath: '',
        informations: [],
      };
    },
  },
  template: `
    <div>
      <BaseButton @click="openModal">Toggle StoreModal</BaseButton>
      <StoreModal
        v-if="ui.isOpen"
        :name="store.name"
        :logoPath="store.logoPath"
        :informations="store.informations"
        :isLoading="ui.isLoading"
        @close="closeModal"
      />
    </div>
  `
})
```

Store modal with link that opens the store page with its own products

```js
new Vue({
  data() {
    return {
      store: {
        name: 'Store example',
        logoPath: 'https://via.placeholder.com/130x74',
        informations: [{
          title: 'Panvel Farmácias',
          content: 'Dimed S/A Distribuidora de Medicamentos<br />CNPJ 92.665.611/0001-77<br />AFE: 7.17094.5 <br />CMVS - 355030801-477-002443-1-7'
        }, {
          title: 'Farmacêutico responsável',
          content: 'Ely Fernandes Lima<br />CRF/PR: 15448'
        }, {
          title: 'Horário de funcionamento',
          content: 'Segunda a Sexta <br />8h00 às 20h30<br />Av. Industrial Belgraf, 865 Eldorado do Sul, Rio Grande do Sul'
        }, {
          title: 'Atendimento',
          content: '(51) 3481-9500<br />sac@grupodimed.com.br'
        }],
      },
      ui: {
        isOpen: false,
      }
    };
  },
  methods: {
    openModal() {
      this.ui.isOpen = true;
    },

    closeModal() {
      this.ui.isOpen = false;
    },
  },
  template: `
    <div>
      <BaseButton @click="openModal">Toggle StoreModal</BaseButton>
      <StoreModal
        v-if="ui.isOpen"
        :name="store.name"
        :logoPath="store.logoPath"
        :informations="store.informations"
        @close="closeModal"
      >
        <div style="margin-left: -18px;">
          <BaseButton type=naked path="#" rel="nofollow">Veja todas as ofertas desta loja</BaseButton>
        </div>
      </StoreModal>
    </div>
  `
})
```
