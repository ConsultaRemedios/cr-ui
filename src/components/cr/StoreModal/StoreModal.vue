<template>
  <StoreBaseModal
    :name="name"
    :logo-path="logoPath"
    :is-loading="isLoading"
    @close="$emit('close')"
  >
    <div :class="$style.content">
      <span :class="$style.title">{{ name }} é uma loja confiável!</span>
      <span :class="$style.subTitle">
        O CR faz uma análise rigorosa para exibir somente lojas confiáveis.
      </span>
      <div :class="$style.wrapperTitleCta">
        <BaseIcon
          :id="icons.truckIcon.id"
          :class="$style.truckIcon"
        />
        <span :class="$style.info">
          <strong>Economize no frete</strong> adicionando produtos da mesma loja
        </span>
      </div>
      <BaseButton
        @click="onClick"
        :path="storePath"
        :class="$style.baseButton"
        type="naked"
      >
        Ver mais produtos da loja
      </BaseButton>

      <div :class="$style.fields">
        <div
          v-for="(field, i) in informations"
          :key="i"
          :class="$style.field"
        >
          <div>
            <span :class="$style.fieldTitle">{{ field.title }}</span>
            <div
              :class="$style.fieldContent"
              v-html="field.content"
            />
          </div>
        </div>
      </div>
    </div>
    <slot />
  </StoreBaseModal>
</template>

<script>
import StoreBaseModal from '../StoreBaseModal';
import BaseButton from '../../common/BaseButton';
import BaseIcon from '../../common/BaseIcon';
import truckIcon from '../../../icons/truck.icon.svg';

export default {
  name: 'StoreModal',
  components: { StoreBaseModal, BaseButton, BaseIcon },
  props: {
    name: {
      type: String,
      default: '',
    },

    logoPath: {
      type: String,
      default: '',
    },

    informations: {
      type: Array,
      default: () => [],
    },

    isLoading: {
      type: Boolean,
      default: false,
    },

    storePath: {
      type: String,
      default: '',
    },
  },

  computed: {
    icons() {
      return {
        truckIcon,
      };
    },
  },

  methods: {
    onClick() {
      this.$emit('clickShowMoreProducts');
    },
  },
};
</script>

<style module>
  @value screen-sm-min from './../../../styles/variables.css';

  .infoData > p {
    font-size: 14px;
    color: var(--text-secondary);
  }

  .subTitle {
    display: block;
    font-size: 14px;
    color : var(--text-default);
    margin-bottom: 34px;
  }

  .title {
    display: block;
    font-size: 20px;
    color: var(--text-default);
    margin-top: 10px;
    margin-bottom: 3px;
    line-height: 20px;
  }

  .baseButton {
    width: 100%;
    height: 44px;
    margin-top: 8px;
    margin-bottom: 20px;
    display: flex;
    justify-content: center;
    align-items: center;
    background: var(--background-color-primary);
    border-color: var(--border-color-primary);
  }

  .baseButton:focus {
    background: var(--background-color-primary-hover);
    border-color: var(--border-color-primary-hover);
  }

  .baseButton span {
    color: white;
    font-size: 16px;
    font-weight: normal;
  }

  .baseButton:hover {
    background: var(--background-color-primary-hover);
    border-color: var(--border-color-primary-hover);
  }

  .addressTitle {
    display: block;
    font-size: 14px;
    color: var(--text-secondary);
  }

  .titleInfos {
    display: block;
    font-size: 14px;
    font-weight: normal;
  }

  .info {
    font-size: 14px;
    color: var(--text-success);
  }

  .field {
    margin-bottom: 32px;
  }

  .field:last-child {
    margin-bottom: 0;
  }

  .fieldTitle {
    font-weight: bold;
    color: var(--text-default);
    display: block;
    margin-bottom: 2px;
    font-size: 14px;
  }

  .fieldContent, .fieldContent > p {
    color: var(--text-default);
    font-weight: normal;
    margin: 0;
    font-size: 14px;
  }

  .fieldContent a {
    color: var(--link-color);
  }

  .fieldContent p:last-child {
    margin-bottom: 0;
  }

  .wrapperTitleCta {
    display: flex;
    align-items: center;
  }

  .truckIcon {
    width: 20px;
    height: 20px;
    color: var(--text-success);
    margin-right: 8px;
  }

  @media (min-width: screen-sm-min) {
    .field {
      display: inline-block;
      width: 100%;
    }
  }
</style>
