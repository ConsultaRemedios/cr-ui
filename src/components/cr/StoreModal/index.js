import StoreModal from './StoreModal.vue';

export default StoreModal;
