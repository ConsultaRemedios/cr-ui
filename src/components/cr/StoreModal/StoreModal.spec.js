import { shallowMount } from '@vue/test-utils';
import { BaseButton } from 'cr-ui';
import StoreModal from './StoreModal.vue';

const shallowStoreModal = (data = {}, slot = []) => {
  return shallowMount(StoreModal, {
    propsData: { ...data },
    slots: {
      default: slot,
    }
  });
};
const informations = [{
  title: 'Panvel Farmácias',
  content: 'Dimed S/A Distribuidora de Medicamentos<br />CNPJ 92.665.611/0001-77<br />AFE: 7.17094.5 <br />CMVS - 355030801-477-002443-1-7'
}, {
  title: 'Farmacêutico responsável',
  content: 'Ely Fernandes Lima<br />CRF/PR: 15448'
}, {
  title: 'Horário de funcionamento',
  content: 'Segunda a Sexta <br />8h00 às 20h30<br />Av. Industrial Belgraf, 865 Eldorado do Sul, Rio Grande do Sul'
}, {
  title: 'Atendimento',
  content: '(51) 3481-9500<br />sac@grupodimed.com.br'
}];

describe('StoreModal component', () => {
  describe('Snapshot', () => {

    describe('When has "name", "logoPath" and "informations"', () => {
      let wrapper;

      beforeAll(() => {
        wrapper = shallowStoreModal({
          name: 'Name',
          logoPath: '/logo/path',
          informations,
        });
      });

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot();
      });
    });

    describe('When "isLoading" is "true"', () => {
      let wrapper;

      beforeAll(() => {
        wrapper = shallowStoreModal({
          isLoading: true,
        });
      });

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot();
      });
    });
  });

  describe('Behaviours', () => {
    it('emits "clickShowMoreProducts" when user clicks in button', () => {
      const wrapper = shallowStoreModal({
        name: 'Name',
        logoPath: '/logo/path',
        informations,
      });

      jest.spyOn(wrapper.vm, '$emit');

      wrapper.findComponent(BaseButton).vm.$emit('click');

      expect(wrapper.vm.$emit).toHaveBeenCalledTimes(1);
      expect(wrapper.vm.$emit).toHaveBeenCalledWith('clickShowMoreProducts');
    });
  });
});
