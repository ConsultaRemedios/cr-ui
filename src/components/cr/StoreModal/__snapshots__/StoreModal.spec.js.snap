// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreModal component Snapshot When "isLoading" is "true" matches 1`] = `
<storebasemodal-stub
  isloading="true"
  logopath=""
  name=""
>
  <div>
    <span
      class="title"
    >
       é uma loja confiável!
    </span>
     
    <span
      class="subTitle"
    >
      
      O CR faz uma análise rigorosa para exibir somente lojas confiáveis.
    
    </span>
     
    <div
      class="wrapperTitleCta"
    >
      <baseicon-stub
        class="truckIcon"
        id="truck.icon.svg"
      />
       
      <span
        class="info"
      >
        <strong>
          Economize no frete
        </strong>
         adicionando produtos da mesma loja
      
      </span>
    </div>
     
    <basebutton-stub
      class="baseButton"
      path=""
      size="medium"
      type="success"
    >
      
      Ver mais produtos da loja
    
    </basebutton-stub>
     
    <div />
  </div>
   
</storebasemodal-stub>
`;

exports[`StoreModal component Snapshot When has "name", "logoPath" and "informations" matches 1`] = `
<storebasemodal-stub
  logopath="/logo/path"
  name="Name"
>
  <div>
    <span
      class="title"
    >
      Name é uma loja confiável!
    </span>
     
    <span
      class="subTitle"
    >
      
      O CR faz uma análise rigorosa para exibir somente lojas confiáveis.
    
    </span>
     
    <div
      class="wrapperTitleCta"
    >
      <baseicon-stub
        class="truckIcon"
        id="truck.icon.svg"
      />
       
      <span
        class="info"
      >
        <strong>
          Economize no frete
        </strong>
         adicionando produtos da mesma loja
      
      </span>
    </div>
     
    <basebutton-stub
      class="baseButton"
      path=""
      size="medium"
      type="success"
    >
      
      Ver mais produtos da loja
    
    </basebutton-stub>
     
    <div>
      <div
        class="field"
      >
        <div>
          <span
            class="fieldTitle"
          >
            Panvel Farmácias
          </span>
           
          <div
            class="fieldContent"
          >
            Dimed S/A Distribuidora de Medicamentos
            <br />
            CNPJ 92.665.611/0001-77
            <br />
            AFE: 7.17094.5 
            <br />
            CMVS - 355030801-477-002443-1-7
          </div>
        </div>
      </div>
      <div
        class="field"
      >
        <div>
          <span
            class="fieldTitle"
          >
            Farmacêutico responsável
          </span>
           
          <div
            class="fieldContent"
          >
            Ely Fernandes Lima
            <br />
            CRF/PR: 15448
          </div>
        </div>
      </div>
      <div
        class="field"
      >
        <div>
          <span
            class="fieldTitle"
          >
            Horário de funcionamento
          </span>
           
          <div
            class="fieldContent"
          >
            Segunda a Sexta 
            <br />
            8h00 às 20h30
            <br />
            Av. Industrial Belgraf, 865 Eldorado do Sul, Rio Grande do Sul
          </div>
        </div>
      </div>
      <div
        class="field"
      >
        <div>
          <span
            class="fieldTitle"
          >
            Atendimento
          </span>
           
          <div
            class="fieldContent"
          >
            (51) 3481-9500
            <br />
            sac@grupodimed.com.br
          </div>
        </div>
      </div>
    </div>
  </div>
   
</storebasemodal-stub>
`;
